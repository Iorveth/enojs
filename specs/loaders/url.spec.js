const eno = require('enojs');

const examples = {
  'http://www.valid.com':  'http://www.valid.com',
  'https://valid.com':     'https://valid.com',
  'https://www.valid.com': 'https://www.valid.com',
  'invalid':               false,
  'www.invalid':           false,
  'www.invalid.com':       false,
  'htp://www.invalid.com': false,
  'http:/invalid.com':     false,
  'https//invalid.com':    false,
  'https://invalid':       false
};

describe('url loader', () => {
  for(let [value, result] of Object.entries(examples)) {
    describe(value, () => {
      const document = eno.parse(`value: ${value}`);

      if(result === false) {
        it('throws an error', () => {
          expect(() => document.url('value')).toThrowErrorMatchingSnapshot();
        });
      } else {
        it(`returns '${result}'`, () => {
          expect(document.url('value')).toBe(result);
        });
      }
    });
  }
});
