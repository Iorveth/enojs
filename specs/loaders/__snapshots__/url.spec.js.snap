// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`url loader htp://www.invalid.com throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: htp://www.invalid.com
"
`;

exports[`url loader http:/invalid.com throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: http:/invalid.com
"
`;

exports[`url loader https//invalid.com throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: https//invalid.com
"
`;

exports[`url loader https://invalid throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: https://invalid
"
`;

exports[`url loader invalid throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: invalid
"
`;

exports[`url loader www.invalid throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: www.invalid
"
`;

exports[`url loader www.invalid.com throws an error 1`] = `
"'value' must contain a valid URL, for instance 'https://eno-lang.org'.

   Line | Content
 >    1 | value: www.invalid.com
"
`;
